controllers/machineset_controller.go

```diff
@@ -161,26 +161,19 @@ func (r *MachineSetReconciler) reconcile(ctx context.Context, machineSet *cluste
 	machineSet.Spec.Selector.MatchLabels[clusterv1.MachineSetLabelName] = machineSet.Name
 	machineSet.Spec.Template.Labels[clusterv1.MachineSetLabelName] = machineSet.Name
 
-	// Make sure that label selector can match template's labels.
-	// TODO(vincepri): Move to a validation (admission) webhook when supported.
 	selector, err := metav1.LabelSelectorAsSelector(&machineSet.Spec.Selector)
 	if err != nil {
 		return ctrl.Result{}, errors.Wrapf(err, "failed to parse MachineSet %q label selector", machineSet.Name)
 	}
-
-	if !selector.Matches(labels.Set(machineSet.Spec.Template.Labels)) {
-		return ctrl.Result{}, errors.Errorf("failed validation on MachineSet %q label selector, cannot match any machines ", machineSet.Name)
-	}
+	// Copy label selector to its status counterpart in string format.
```

Comment:
nit: this didn't need to move, did it?

Reply:
It didn't. When I was refactoring I thought it would make sense to move the only usage of `selector` soon after it was created.

Reply:
That's fair. When I review, one thing I look out for is keeping the change set to a minimum, so I ask questions like this one. Thank you for pointing out your reasoning. It's fine to move it so it's closer. Thanks.